### PR TITLE
[recipe][pycppad] add new recipe for pycppad

### PR DESCRIPTION
**Pull request by proyan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by proyan** [comment]
Since this file provides python bindings written in C++ using boost python, @conda-forge/help-c-cpp or @conda-forge/help-python-c could you please provide a review. Thanks.

**Comment by proyan** [comment]
cc: @jcarpent

File: recipes/pycppad/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  version: {{ version }}
+
+source:
+
+  url: https://github.com/Simple-Robotics/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8e3da0798a69b275a1cd0574939553fc65b152333e18832a791cfad199da0378
+
+build:
+  number: 0
+  skip: true  # [win and vc<14]
```
Comment (xhochy) [inline recipes/pycppad/meta.yaml:15]:
```suggestion
```

File: recipes/pycppad/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [not win]
+    - vs2017_win-64  # [win64]
+    - vs2017_win-32  # [win32]
+    - cmake
```
Comment (xhochy) [inline recipes/pycppad/meta.yaml:22]: All point to the same
```suggestion
    - {{ compiler('cxx') }}
    - cmake
```

File: recipes/pycppad/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - eigen
+    - boost
+    - python
+    - cppad
+    - {{ pin_compatible('numpy') }}
+
+test:
+  requires:
+    - pkg-config
+    - boost
```
Comment (xhochy) [inline recipes/pycppad/meta.yaml:43]: Is already part of run
```suggestion
```

**Comment by jcarpent** [comment]
> cc: @jcarpent

I agree to be a maintainer of this package.

File: recipes/pycppad/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - make  # [not win]
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:23]:
```suggestion
```

File: recipes/pycppad/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  host:
+    - eigen
+    - boost
+    - python
+    - numpy
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:28]:
```suggestion
```

File: recipes/pycppad/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    - cppad
+  run:
+    - eigen
+    - boost
+    - python
+    - cppad
+    - {{ pin_compatible('numpy') }}
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:35]:
```suggestion
```

File: recipes/pycppad/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    - make  # [not win]
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+  host:
+    - eigen
+    - boost
+    - python
+    - numpy
+    - cppad
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:29]:
```suggestion
    - cppad
    - eigenpy
```

File: recipes/pycppad/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  run:
+    - eigen
+    - boost
+    - python
+    - cppad
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:34]:
```suggestion
    - cppad
    - eigenpy
```

File: recipes/pycppad/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  summary: Python bindings for CppAD and CppADCodeGen using Boost.Python 
+  dev_url: http://github.com/Simple-Robotics/pycppad
+
+extra:
+  recipe-maintainers:
+    - proyan
+    - jcarpent
+
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:57]:
```suggestion
    
    
```

File: recipes/pycppad/bld.bat [hunk 10]
```diff
@@ -0,0 +1,21 @@
+set PKG_CONFIG_PATH=%LIBRARY_PREFIX%\share\pkgconfig
```
Comment (jcarpent) [inline recipes/pycppad/bld.bat:1]:
```suggestion
```

File: recipes/pycppad/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+
+  url: https://github.com/Simple-Robotics/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8e3da0798a69b275a1cd0574939553fc65b152333e18832a791cfad199da0378
+
+build:
+  number: 0
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:14]:
```suggestion
  number: 0
  skip: true  # [win]
```

**Review by jcarpent — CHANGES_REQUESTED** [review]
I made some suggestions which should help the linter and make the project compile

File: recipes/pycppad/bld.bat (hunk 10 above)
Comment (jcarpent) [reply recipes/pycppad/bld.bat:1]: useless

File: recipes/pycppad/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  summary: Python bindings for CppAD and CppADCodeGen using Boost.Python 
+  dev_url: http://github.com/Simple-Robotics/pycppad
+
+extra:
+  recipe-maintainers:
+    - proyan
+    - jcarpent
+
```
Comment (jcarpent) [inline recipes/pycppad/meta.yaml:57]:
```suggestion
```

Comment (jcarpent) [reply recipes/pycppad/meta.yaml:57]:
```suggestion
```

**Comment by proyan** [comment]
@xhochy The linting is passing now :)

**Comment by xhochy** [comment]
The build is failing though: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=365986&view=results

**Comment by proyan** [comment]
I've made a new release that should pass. Could you please launch the build again?

**Comment by jcarpent** [comment]
@proyan you need to change the version and sha 256 to account for the new release.

**Comment by proyan** [comment]
Thanks for the ping

**Comment by jcarpent** [comment]
Currently, there are some troubles to sync the CI with GitHub on conda builds.
See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366252&view=results for the current status

**Comment by jcarpent** [comment]
Good news. The build is passing.
@conda-forge/help-c-cpp @conda-forge/help-python this PR is ready for being merged.

**Pull request merged**